train.py

```diff
@@ -24,8 +24,14 @@
 incorrect_id = 0
 correct_id = 1
 
-def generate_data(max_neighbors):
-    for csv_file in preprocessed_path.glob('*.csv'):
+def generate_data(max_neighbors, train=True, test_size=0.1):
```

Comment:
`glob()` calls the underlying OS's listdir function, which doesn't always give back results in a consistent order. If we call this twice and it returns different orders, we run the risk of data leakage between train and test. Option 1 is to sort the glob results here alphabetically, then randomly shuffle (with an explicit seed number, for reproducibility), then split. Option 2 is to do all of that in `__main__`, and pass in a list of files to `generate_data()`